internal/util/util.go

```diff
@@ -7,21 +7,76 @@ import (
 	"strings"
 )
 
-// StructToStruct converts a struct type (src) into another (dst)
-func StructToStruct(src any, dst any) error {
+func JsonUnmarshaler(src any, dst any) error {
 	jsonBytes, err := json.Marshal(src)
 	if err != nil {
 		return err
 	}
 
-	err = json.Unmarshal(jsonBytes, dst)
-	if err != nil {
+	if err = json.Unmarshal(jsonBytes, dst); err != nil {
 		return fmt.Errorf("failed to convert json %s: err=%s", string(jsonBytes), err)
 	}
-
 	return nil
 }
 
+// StructToStruct converts a struct to a map[string]interface{} by marshalling the struct to JSON and then
+// unmarshalling it to a map. It also converts all json.Number values to int64 or float64.
+func StructToStruct(src any) (map[string]any, error) {
+	jsonBytes, err := json.Marshal(src)
+	if err != nil {
+		return nil, err
+	}
+
+	var parsedValue map[string]any
+	d := json.NewDecoder(strings.NewReader(string(jsonBytes)))
+	d.UseNumber()
+	err = d.Decode(&parsedValue)
+	if err != nil {
+		return parsedValue, fmt.Errorf("failed to convert json %s: err=%s", string(jsonBytes), err)
+	}
+	convertNumbers(parsedValue)
+
+	return parsedValue, nil
+}
+
+func convertNumbers(parsedValue map[string]any) {
+	for k, v := range parsedValue {
+		switch t := v.(type) {
+		case json.Number:
+			parsedValue[k] = convertNumber(t)
+		case map[string]any:
+			convertNumbers(t)
+		case []any:
+			convertNumbersArray(t)
+		}
+	}
+}
+
+func convertNumbersArray(arr []any) {
+	for i, v := range arr {
+		switch t := v.(type) {
+		case json.Number:
+			arr[i] = convertNumber(t)
+		case map[string]any:
+			convertNumbers(t)
+		case []any:
+			convertNumbersArray(t)
+		}
+	}
+}
+
+func convertNumber(value json.Number) any {
+	i64, err := value.Int64()
```

Comment:
It looks like this will return an Int64 if it can be parsed as an Int64. But would that work in all cases?

Let's say hypothetically that the `Generation` field were actually supposed to be a float64. But in the json it happens to be a "4". Thus, it can be parsed as an Int64. But then later we call `GetGeneration()` and the type is an Int64 instead of a Float64. It will fail.